src/jit/codegenarmarch.cpp

```diff
@@ -1621,7 +1769,21 @@ void CodeGen::genCallInstruction(GenTreeCall* call)
             genConsumeReg(argNode);
             if (argNode->gtRegNum != argReg)
```

Comment:
I think that this line should be moved to the `else` clause below, and in the `isSplit` case, each register needs to be consumed.